testing/src/main/scala/prometheus4cats/testing/TestingMetricRegistry.scala

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright 2022 Permutive
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package prometheus4cats
+package testing
+
+import cats.syntax.all._
+import cats.data.Chain
+import cats.effect.kernel._
+import cats.effect.std.MapRef
+import cats.data.NonEmptySeq
+import prometheus4cats.util.DoubleMetricRegistry
+import scala.concurrent.duration.FiniteDuration
+import TestingMetricRegistry._
+
+class TestingMetricRegistry[F[_]](
```

Comment:
I think this constructor could do with being private so that the data structure can change in the future if needed.

I realise this would mean it's harder to get at the underlying `MapRef`s, but I wonder if it's possible to expose the metric collections in a generic way that doesn't tie us to the implementation. I release that might conflict with my earlier review comment asking you to expose the underlying data, so if it's too much of a pain exposing the data, then I'd say not bother with lower level access and just provide the higher level methods.

Reply:
Yeah maybe let's start with just exposing our higher level methods and see if people come up with other usecases

Reply:
Yep, that makes sense. It's much easier to add with bincompat, even if it's class that extends this. Which makes me wonder, should this be `sealed abstract`?

Reply:
Yeah makes sense 👍